app/dependencies.py

```diff
@@ -0,0 +1,32 @@
+from typing import Optional
+
+from fastapi import Header, Request
+
+from app.api_tokens import validate_api_token
+
+
+async def verify_api_token(
+    request: Request,
+    x_api_key: Optional[str] = Header(
+        None, title="API token for accessing this source"
+    ),
+    source: Optional[str] = "",
+):
+    """
+    The "source" is expected as a query parameter ?source=example
+    The request path is expected to be the data product path, if you have some other base path, make sure you trim it.
+
+    :param request:
+    :param x_api_key:
+    :param source:
+    :raises Exception: In case validation fails
+    """
```

Comment:
Could we improve this, possibly something like this (feel free to modify according to own preferences):


```suggestion
    """
    A FastAPI dependency that will validate the "X-API-KEY" if it's present, but also allows requests without it to demonstrate how requests without any API tokens work. Please note that if you plan to base your API token verification upon this code.

    :param request: The request path is expected to be the data product path, if you have some other base path, make sure you trim it.
    :param x_api_key: The X-API-KEY header to (optionally) be verified.
    :param source: The "source" is expected as a query parameter ?source=example
    :raises Exception: In case validation fails
    """
```

Reply:
I took your suggestion as a base and modified it somewhat.